### ISSUE TITLE
25000 context sanity limit hit for PHP syntax

### ISSUE THREAD

**Issue by BokuNoCode** [borela/naomi#183]
25000 context sanity limit hit for PHP syntax

**Comment by beznosd** [comment]
Just after installation of Naomi experienced the same issue.
<img width="417" alt="context-limit" src="https://user-images.githubusercontent.com/8027160/47823749-5bbe2b80-dd72-11e8-8b90-2d3b75b6297d.png">

**Comment by borela** [comment]
This is being fixed https://github.com/borela/naomi/issues/181#issuecomment-431522115

**Comment by beznosd** [comment]
Thank you!

**Comment by chavab1** [comment]
> Avoid using the builtin PHP, Markdown and Latex syntaxes if possible

Are we suppose to remove the default PHP syntax temporarily??

**Comment by borela** [comment]
> 
> 
> > Avoid using the builtin PHP, Markdown and Latex syntaxes if possible
> 
> Are we suppose to remove the default PHP syntax temporarily??

No, you just configure files to open with the alternative syntax by default.

**Comment by chavab1** [comment]
OK, thanx for the clarification and your help.

**Comment by brokeyourbike** [comment]
Is there a way to disable only Naomi PHP syntax?

**Comment by borela** [comment]
@brokeyourbike 

Rename the “naomi.php7.sublime-syntax” to ““naomi.php7.sublime-syntax.disabled” located at:

* **OS X**: `~/Library/Application Support/Sublime Text 3/Packages/Naomi/syntaxes/`
* **Linux**: `~/.config/sublime-text-3/Packages/Naomi/syntaxes/`
* **Windows**: `%APPDATA%\Sublime Text 3\Packages\Naomi\syntaxes\`

**Comment by cibulka** [comment]
@borela Unfortunately that does not help.

 I renamed the file `/Users/<USER>/Library/Application Support/Sublime Text 3/Packages/Naomi/syntaxes\naomi.php7.sublime-syntax` to `/Users/<USER>/Library/Application Support/Sublime Text 3/Packages/Naomi/syntaxes
naomi.php7.sublime-syntax.disabled`, restarted Sublime text and error persists: `Error loading syntax file "Packages/PHP/PHP.sublime-syntax": Apparent recursion within a with_prototype action: 25000 context sanity limit hit`. 

What did I do wrong?

**Comment by rishisharma08** [comment]
naomi's php syntax highlighting breaks when there are escaped single quotes inside a string of single quotes.
`$str = 'asd\'s \'asd';`
 so i was trying to revert back to native php syntax highlighter but that broke. so i ended up disabling naomi for now. i downloaded it for React but ill just have to make do without it for now.

**Comment by mflash123** [comment]
Still waiting for a fix

**Comment by humarh-dharnarh** [comment]
> @borela Unfortunately that does not help.
> 
> I renamed the file `/Users/<USER>/Library/Application Support/Sublime Text 3/Packages/Naomi/syntaxes\naomi.php7.sublime-syntax` to `/Users/<USER>/Library/Application Support/Sublime Text 3/Packages/Naomi/syntaxes naomi.php7.sublime-syntax.disabled`, restarted Sublime text and error persists: `Error loading syntax file "Packages/PHP/PHP.sublime-syntax": Apparent recursion within a with_prototype action: 25000 context sanity limit hit`.
> 
> What did I do wrong?

Were you able to disable it yet? doesn't seem to work for me

**Comment by devnix** [comment]
Just tried this plugin. In first place, it broke the original PHP syntax, and when I've selected the Naomi PHP syntax, it didn't highlight HTML outside the PHP tags. Loved how it displays pure PHP files though, but it's unusable for me

**Comment by CptTony** [comment]
Welp, looks like this isn't getting fixed any time soon. Gotta find another plugin for react cause Naomi is no good for php and breaks the native highlighting.

**Comment by jeremytm** [comment]
Same as @rishisharma08, installed for react but it breaks php... Why wouldn't they fix it? Is this project abandoned?

**Comment by maerzhase** [comment]
+1

**Comment by KernelZechs** [comment]
Attached is a working plugin that supports Naomi PHP7, Sublime PHP7, and React/JSX using Naomi without conflict.

FWIW, the breaking syntax highlighter was `css3` which, NGL, seems useless since sublimes CSS highlighter is perfectly fine.

Just delete the original package and replace it with this one. For good measure close your sublime, delete your cache folder, reopen.

[Naomi-WorkingReact-PHP.tar.gz](https://github.com/borela/naomi/files/5751323/Naomi-WorkingReact-PHP.tar.gz)

**Comment by boycce** [comment]
@KernelZechs this seems to break React CSS (style-components)

**Comment by KernelZechs** [comment]
> @KernelZechs removing seems to break React CSS (style-components)

It does, however I don't use them (and in the work capacity where I do we're migrating away from them) so the above solution works fine for me. I can see if I can try to make another patch to fix that problem, but I wish the Naomi team would actually address this ticket instead of me championing this cause.

**Comment by h0jeZvgoxFepBQ2C** [comment]
Please fix this, Team!
